content/blog/2022-04-26-april-22-community-gems.md

```diff
@@ -0,0 +1,124 @@
+---
+title: April '22 Community Gems
+date: 2022-04-26
+description: >
+  A roundup of technical Q&A's from the DVC and CML community. This month: CML
+  updates, working with multiple datasets, using DVC stages, and more.
+descriptionLong: >
+  A roundup of technical Q&A's from the DVC and CML community. This month: CML
+  updates, working with multiple datasets, using DVC stages, and more.
+picture: 2022-04-326/apr-comm-gems.png
+author: milecia_mcgregor
+commentsUrl:
+tags:
+  - CML
+  - DVC Stages
+  - DVC Remotes
+  - Community
+---
+
+### [When I run `dvc repro` on a stage, does it automatically push any outputs to my remote?](https://discord.com/channels/485586884165107732/563406153334128681/953616587523498025)
+
+Great question from @tina_rey!
+
+The `dvc repro` command doesn't automatically push any outputs or data to your
+remote. The outputs are stored in the cache until you run `dvc push`, which then
+pushes them from your cache to your remote.
+
+### [Is `dvc dag` based on `deps` and `outs`, so that a stage that depends on the output of another stage will always be executed after the former has finished?](https://discord.com/channels/485586884165107732/563406153334128681/956113493155799070)
+
+This is a good question from @johnysku!
+
+That is correct! If the pipelines are independent or the stages are independent,
+they may run in any order. Without explicit dependency linkage, stages could be
+executed in an unexpected order.
+
+One way around this is by outputing some kind of placeholder file when all a
+stage is finished and make that the dependency that triggers the next stage.
```

Comment:
I'm not fully understanding this description. Maybe it's unnecessary. It sounds like the discord discussion went deeper than the initial question into some specific scenario, but it's not really clear how this relates to the original question.

Reply:
Gotcha, thanks for catching this.